### PR TITLE
Improve MirrorNodeClient 500 handling

### LINKED ISSUE #261: MirrorNode Response logic doesn't handle missing status code
### Description

The `MirrorNodeClient` fails when the returned error object is missing a response i.e. `error.response.status`

Call fails with following error and metric and log are missed
```
err: {
      "type": "TypeError",
      "message": "Cannot read properties of undefined (reading 'status')",
      "stack":
          TypeError: Cannot read properties of undefined (reading 'status')
              at MirrorNodeClient.<anonymous> (.../hedera-json-rpc-relay/packages/relay/dist/lib/clients/mirrorNodeClient.js:92:32)
              at Generator.throw (<anonymous>)
              at rejected (.../hedera-json-rpc-relay/packages/relay/dist/lib/clients/mirrorNodeClient.js:25:65)
              at processTicksAndRejections (node:internal/process/task_queues:96:5)
    }
```

### Steps to reproduce

1. Make a relay call that results in a mirror node call that may fail
2. Expect a valid status Code to be picked up

### Additional context

_No response_

### Hedera network

testnet

### Version

v0.2.0-rc1

### Operating system

_No response_

### PR DESCRIPTION
**Pull request by Nana-EC**
Signed-off-by: Nana-EC <nana@swirldslabs.com>

**Description**:
Improve 500 response handling
- Update `MirrorNodeClient` to check for valid error response and handle absence

**Related issue(s)**:

Fixes #261 

**Notes for reviewer**:
Flow now correctly displays
```
err: {
  "type": "Error",
  "message": "connect ECONNREFUSED 127.0.0.1:5551",
  "stack":
      Error: connect ECONNREFUSED 127.0.0.1:5551
          at MirrorNodeClient.handleError (.../hedera-json-rpc-relay/packages/relay/dist/lib/clients/mirrorNodeClient.js:106:27)
          at MirrorNodeClient.<anonymous> (.../hedera-json-rpc-relay/packages/relay/dist/lib/clients/mirrorNodeClient.js:94:22)
          at Generator.throw (<anonymous>)
          at rejected (.../hedera-json-rpc-relay/packages/relay/dist/lib/clients/mirrorNodeClient.js:25:65)
          at processTicksAndRejections (node:internal/process/task_queues:96:5)
}

[POST] eth_blockNumber: -32603 6 msESC[39m
code: -32603
message: "Unknown error invoking RPC"
```

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by ar-conmit — DISMISSED** [review]
LGTM

File: packages/relay/src/lib/clients/mirrorNodeClient.ts [hunk 1]
```diff
@@ -70,6 +70,8 @@ export class MirrorNodeClient {
         DESC: 'desc'
     };
 
+    private static httpCodeInternalServerError = 500;
```
Comment (shemnon) [inline packages/relay/src/lib/clients/mirrorNodeClient.ts:73]: Maybe a different 500 series, to distinguish between server sent 500s?

**Review by shemnon — APPROVED** [review]

File: packages/relay/src/lib/clients/mirrorNodeClient.ts [hunk 2]
```diff
@@ -70,6 +70,8 @@ export class MirrorNodeClient {
         DESC: 'desc'
     };
 
+    private static badGatewayHttpStatusCode = 502;
```
Comment (shemnon) [inline packages/relay/src/lib/clients/mirrorNodeClient.ts:73]: Does it have to be one that is assigned?  I could see someone accessing mirrornode through a gateway and legitametly getting this.  Maybe 567 as it is obviously not a real server error?

**Review by shemnon — APPROVED** [review]

File: packages/relay/src/lib/clients/mirrorNodeClient.ts (hunk 2 above)
Comment (Nana-EC) [reply packages/relay/src/lib/clients/mirrorNodeClient.ts]: Doesn't have to be, just need to be consistent for metrics.
I'm updated

**Review by shemnon — APPROVED** [review]

**Pull request merged**